Client/Components/DashBoard.razor.cs

```diff
@@ -89,10 +87,53 @@ private static string FormatLabel(string label)
 
         private void HandleFilterChange(IEnumerable<string> selectedValues)
         {
-            _dataMetrics = selectedValues.ToList();
+            var selectedMetrics = selectedValues.ToList();
+
+            Dispatcher.Dispatch(new ShownMetricsSelectedAction(selectedMetrics));
+        }
+
+        private async Task InitHomeCountryAsync()
+        {
+            if (State.Value.HomeCountry != null) return;
 
-            // Makes sure that when you select unselected values, they end up in the same order as before
-            _dataMetrics = _dataMetrics.OrderBy(d => _availableMetrics.IndexOf(d)).ToList();
```

Comment:
Why did this get removed? Metrics that are deselected and reselected don't appear in the correct order anymore?

Reply:
That might've been an oversight on my part. Would it make sense to add it back in before the action is dispatched?

Reply:
I think this line of code should be in the HandleFilterChange() method, probably before the action is dispatched if I understand it correctly.

Reply:
Fixed